pkg/driver/controller_modify_volume.go

```diff
@@ -17,8 +20,147 @@ const (
 	ModificationKeyIOPS = "iops"
 
 	ModificationKeyThroughput = "throughput"
+
+	modifyVolumeRequestHandlerTimeout = 2 * time.Second
 )
 
+type modifyVolumeRequest struct {
+	newSize           int64
+	modifyDiskOptions cloud.ModifyDiskOptions
+}
+
+type modifyVolumeRequestHandler struct {
+	volumeID string
+	// Merged request from the requests that have been accepted for the volume
+	mergedRequest *modifyVolumeRequest
+	// Channel for sending requests to the Timer thread for the volume
+	requestChan chan *modifyVolumeRequest
+	// Channel for sending errors of ec2 ModifyVolume API call to the CSI Driver main thread
+	errChan chan error
+	// Channel for sending the actual volume size after the ec2 ModifyVolume API call to the CSI Driver main thread
+	resultChan chan int64
+	mux sync.RWMutex
+}
+
+type modifyVolumeManager struct {
+	// Map of volume ID to modifyVolumeRequestHandler
+	requestHandlerMap map[string]*modifyVolumeRequestHandler
+	mux               *sync.RWMutex
+}
+
+func newModifyVolumeManager() *modifyVolumeManager {
+	return &modifyVolumeManager{
+		requestHandlerMap: make(map[string]*modifyVolumeRequestHandler),
+		mux:               &sync.RWMutex{},
```

Comment:
Suggest using a  [sync.Map](https://pkg.go.dev/sync#Map) to protect the `requestHandlerMap` here which handles all the locking/synchronization internally, it's optimized for concurrency and will greatly simplify the code by removing the need for explicit lock management.

Reply:
Done